--- threadedcomments/models.py.old	2010-05-13 18:47:15.000000000 +0600
+++ threadedcomments/models.py	2010-12-23 03:12:28.000000000 +0500
@@ -13,9 +13,10 @@
         related_name='children', verbose_name=_('Parent'))
     last_child = models.ForeignKey('self', null=True, blank=True,
         verbose_name=_('Last child'))
-    tree_path = models.TextField(_('Tree path'), editable=False,
+    tree_path = models.CharField(max_length=255, editable=False, blank=True,
         db_index=True)
 
+
     objects = CommentManager()
 
     def _get_depth(self):
