diff -Nuar transifex-0.9.1.orig/transifex/reviews/models.py transifex-0.9.1/transifex/reviews/models.py
--- transifex-0.9.1.orig/transifex/reviews/models.py	2011-02-13 20:24:32.000000000 +0500
+++ transifex-0.9.1/transifex/reviews/models.py	2011-02-13 20:25:27.000000000 +0500
@@ -86,7 +86,7 @@
     class Meta:
         verbose_name = _('Review Request')
         verbose_name_plural = _('Review Requests')
-        ordering  = (_('-created_on'),)
+        ordering  = ('-created_on'),
         get_latest_by = _('created_on')
 
 
