diff -r e7d48205dbe1 transifex/txcommon/listeners.py
--- a/transifex/txcommon/listeners.py	Thu Nov 25 10:25:55 2010 -0200
+++ b/transifex/txcommon/listeners.py	Thu Dec 23 03:06:55 2010 +0500
@@ -23,7 +23,7 @@
     if 'created' in kwargs and kwargs['created'] is True: 
         user = kwargs['instance']
         # Create Public Profile
-        profile, created = Profile.objects.get_or_create(user=user)
+        profile, created = Profile.objects.get_or_create(user=user, location='')
         # Add user to registered group
         group, created = Group.objects.get_or_create(name='registered')
         user.groups.add(group)
