
--- a/transifex/addons/jsonmap/migrations/0001_initial.py	Thu Sep 02 11:49:32 2010 +0300
+++ b/transifex/addons/jsonmap/migrations/0001_initial.py	Fri Oct 22 11:28:07 2010 -0200
@@ -16,7 +16,7 @@
         # Adding model 'JSONMap'
         db.create_table('jsonmap_jsonmap', (
             ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
-            ('slug', self.gf('django.db.models.fields.TextField')(max_length=50)),
+            ('slug', self.gf('django.db.models.fields.SlugField')(max_length=50)),
             ('content', self.gf('txcommon.db.models.CompressedTextField')(null=False, blank=False)),
             ('created', self.gf('django.db.models.fields.DateTimeField')(auto_now_add=True, blank=True)),
             ('modified', self.gf('django.db.models.fields.DateTimeField')(auto_now=True, blank=True)),




