--- xml2po/xml2po.py.orig	2009-05-10 17:42:35.000000000 +0600
+++ xml2po/xml2po.py	2009-05-13 20:48:57.000000000 +0600
@@ -51,7 +51,8 @@
         return None
 
     def ugettext(self, message):
-        return None
+        # not None!
+        return message
 
     def ungettext(self, msgid1, msgid2, n):
         return None
@@ -83,7 +84,7 @@
             if self.output_msgstr:
                 self.translations.append(t)
                 return
-            
+
             if self.do_translations or (not t in self.messages):
                 self.messages.append(t)
                 if spacepreserve:
@@ -520,7 +521,12 @@
         if restart or worth:
             i = 0
             while i < len(myrepl):
-                replacement = '<%s>%s</%s>' % (myrepl[i][0], myrepl[i][3], myrepl[i][2])
+                # unicode -> string
+                replacement = '<%s>%s</%s>' % (myrepl[i][0],
+                                                   myrepl[i][3].encode('utf-8'),
+                                                   myrepl[i][2])
+                # string -> unicode
+                replacement = replacement.decode('utf-8')
                 i += 1
                 translation = translation.replace('<placeholder-%d/>' % (i), replacement)
 
@@ -713,9 +719,9 @@
 import getopt, fileinput
 
 def usage (with_help = False):
-        print >> sys.stderr, "Usage:  %s [OPTIONS] [XMLFILE]..." % (sys.argv[0])
-	if (with_help):
-        	print >> sys.stderr, """
+    print >> sys.stderr, "Usage:  %s [OPTIONS] [XMLFILE]..." % (sys.argv[0])
+    if (with_help):
+            print >> sys.stderr, """
 OPTIONS may be some of:
     -a    --automatic-tags     Automatically decides if tags are to be considered
                                  "final" or not
@@ -746,7 +752,7 @@
         %s -p de.po chapter1.xml > chapter1.de.xml
         %s -p de.po chapter2.xml > chapter2.de.xml
 """ % (sys.argv[0], sys.argv[0], sys.argv[0])
-        sys.exit(0)
+    sys.exit(0)
 
 if len(sys.argv) < 2: usage()
 
@@ -789,7 +795,7 @@
         print VERSION
         sys.exit(0)
     elif opt in ('-h', '--help'):
-    	usage(True)
+        usage(True)
 
 # Treat remaining arguments as XML files
 while args:
