Index: SConstruct
===================================================================
--- SConstruct	(revision 700)
+++ SConstruct	(working copy)
@@ -797,7 +797,7 @@
         Alias('install',
               env.Command(lib_dir + '/' + gigi_libname,
                           lib_dir + '/' + installed_gigi_libname,
-                          'ln -s ' + lib_dir + '/' + installed_gigi_libname + ' ' + lib_dir + '/' + gigi_libname))
+                          'ln -s ' + installed_gigi_libname + ' ' + lib_dir + '/' + gigi_libname))
 if not missing_pkg_config and str(Platform()) != 'win32':
     Alias('install', Install(env.subst(env['pkgconfigdir']), env.File('GiGi.pc')))
 if env['build_sdl_driver']:
@@ -809,7 +809,7 @@
             Alias('install',
                   env.Command(lib_dir + '/' + gigi_sdl_libname,
                               lib_dir + '/' + installed_gigi_sdl_libname,
-                              'ln -s ' + lib_dir + '/' + installed_gigi_sdl_libname + ' ' + lib_dir + '/' + gigi_sdl_libname))
+                              'ln -s ' + installed_gigi_sdl_libname + ' ' + lib_dir + '/' + gigi_sdl_libname))
     if not missing_pkg_config and str(Platform()) != 'win32':
         Alias('install', Install(env.subst(env['pkgconfigdir']), env.File('GiGiSDL.pc')))
 if env['build_ogre_driver']:
@@ -821,7 +821,7 @@
             Alias('install',
                   env.Command(lib_dir + '/' + gigi_ogre_libname,
                               lib_dir + '/' + installed_gigi_ogre_libname,
-                              'ln -s ' + lib_dir + '/' + installed_gigi_ogre_libname + ' ' + lib_dir + '/' + gigi_ogre_libname))
+                              'ln -s ' + installed_gigi_ogre_libname + ' ' + lib_dir + '/' + gigi_ogre_libname))
     if not missing_pkg_config and str(Platform()) != 'win32':
         Alias('install', Install(env.subst(env['pkgconfigdir']), env.File('GiGiOgre.pc')))
 if env['build_ogre_driver'] and env['build_ogre_ois_plugin']:
