diff -Nuar uftp-4.1.4.orig/makefile uftp-4.1.4/makefile
--- uftp-4.1.4.orig/makefile	2014-01-19 14:12:20.016853178 +0600
+++ uftp-4.1.4/makefile	2014-01-19 14:15:42.652860174 +0600
@@ -42,26 +42,27 @@
 endif
 
 # defaults
-CC = gcc
-OPTIONS=-g -Wall $(ENC_OPTS)
+CC ?= gcc
+OPTIONS=$(ENC_OPTS)
 LDLIBS=-lc -lm $(CRYPT_LIB)
-CFLAGS=
+CFLAGS?=
+LDFLAGS?=
 MTFLAGS=
 
 # FreeBSD
 ifeq ("FreeBSD", "$(UNAME_S)")
-OPTIONS=-g -Wall -DHAS_GETIFADDRS -DNO_DUAL -DNO_MCAST_JOIN $(ENC_OPTS)
+OPTIONS=-DHAS_GETIFADDRS -DNO_DUAL -DNO_MCAST_JOIN $(ENC_OPTS)
 endif
 
 # OSX, aka Darwin
 ifeq ("Darwin", "$(UNAME_S)")
-OPTIONS=-g -Wall -DHAS_GETIFADDRS $(ENC_OPTS)
+OPTIONS=-DHAS_GETIFADDRS $(ENC_OPTS)
 endif
 
 # Sun
 ifeq ("SunOS", "$(UNAME_S)")
 CC = cc
-OPTIONS=-g -DBSD_COMP -DNO_DUAL $(ENC_OPTS)
+OPTIONS=-DBSD_COMP -DNO_DUAL $(ENC_OPTS)
 LDLIBS=-lnsl -lsocket -lm $(CRYPT_LIB)
 CFLAGS=`getconf LFS_CFLAGS`
 OPENSSL=/usr/sfw
@@ -70,9 +71,9 @@
 
 # Linux
 ifeq ("Linux", "$(UNAME_S)")
-OPTIONS=-g -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wformat=2 -Wwrite-strings -Wpointer-arith -Wcast-qual -Wshadow -Wno-missing-field-initializers -Wstrict-prototypes -Winline -Wbad-function-cast -DHAS_GETIFADDRS $(ENC_OPTS)
+OPTIONS=-Wextra -Wno-unused-parameter -Wno-sign-compare -Wformat=2 -Wwrite-strings -Wpointer-arith -Wcast-qual -Wshadow -Wno-missing-field-initializers -Wstrict-prototypes -Winline -Wbad-function-cast -DHAS_GETIFADDRS $(ENC_OPTS)
 LDLIBS=-lm $(CRYPT_LIB)
-CFLAGS=`getconf LFS_CFLAGS`
+CFLAGS?=`getconf LFS_CFLAGS`
 endif
 
 ifdef OPENSSL
@@ -87,7 +88,7 @@
 testclient_multi.o: testclient_multi.c uftp.h uftp_common.h encryption.h
 
 testclient_multi: testclient_multi.o encrypt_openssl.o uftp_common.o
-	$(CC) $(OPTIONS) $(LIB) -o $@ $^ $(LDLIBS)
+	$(CC) $(OPTIONS) $(LIB) -o $@ $^ $(LDLIBS) $(LDFLAGS)
 
 clean:
 	rm -f testclient_multi uftp uftpd uftpproxyd uftp_keymgt *.o
@@ -182,16 +183,16 @@
 endif
 
 uftp: $(UFTP_OBJS)
-	$(CC) $(OPTIONS) $(LIB) -o $@ $^ $(LDLIBS) $(MTFLAGS) -lpthread
+	$(CC) $(OPTIONS) $(LIB) -o $@ $^ $(LDLIBS) $(LDFLAGS) $(MTFLAGS) -lpthread
 
 uftpd: $(UFTPD_OBJS)
-	$(CC) $(OPTIONS) $(LIB) -o $@ $^ $(LDLIBS)
+	$(CC) $(OPTIONS) $(LIB) -o $@ $^ $(LDLIBS) $(LDFLAGS)
 
 uftpproxyd: $(UFTPPROXYD_OBJS)
-	$(CC) $(OPTIONS) $(LIB) -o $@ $^ $(LDLIBS)
+	$(CC) $(OPTIONS) $(LIB) -o $@ $^ $(LDLIBS) $(LDFLAGS)
 
 uftp_keymgt: $(UFTP_KEYMGT_OBJS)
-	$(CC) $(OPTIONS) $(LIB) -o $@ $^ $(LDLIBS)
+	$(CC) $(OPTIONS) $(LIB) -o $@ $^ $(LDLIBS) $(LDFLAGS)
 
 %.o: %.c
 	$(CC) $(OPTIONS) $(INCLUDE) $(CFLAGS) $(MTFLAGS) -c $<
