From fa9042fea0e98aed0fb63aa2d43e5f637f10dad8 Mon Sep 17 00:00:00 2001
From: "Azamat H. Hackimov" <azamat.hackimov@gmail.com>
Date: Thu, 16 Jun 2022 23:04:30 +0300
Subject: [PATCH] Adding support of LDFLAGS

--- a/Makefile
+++ b/Makefile
@@ -21,12 +21,12 @@ INSTALL_DATA = $(INSTALL) -m 644
 all: recoil2png $(if $(CAN_INSTALL_MAGICK),imagemagick/recoil.so) recoil-mime.xml
 
 recoil2png: recoil2png.c pngsave.c pngsave.h recoil-stdio.c recoil-stdio.h recoil.c recoil.h
-	$(CC) $(CFLAGS) recoil2png.c pngsave.c recoil-stdio.c recoil.c -lpng -lz -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) recoil2png.c pngsave.c recoil-stdio.c recoil.c -lpng -lz -o $@
 
 ifdef CAN_INSTALL_MAGICK
 imagemagick/recoil.so: imagemagick/recoilmagick.c recoil.c recoil.h formats.h
 ifdef MAGICK_INCLUDE_PATH
-	$(CC) $(CFLAGS) $(MAGICK_CFLAGS) -I$(MAGICK_INCLUDE_PATH) imagemagick/recoilmagick.c recoil.c -shared $(MAGICK_LDFLAGS) -ldl $(MAGICK_LIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $(MAGICK_CFLAGS) -I$(MAGICK_INCLUDE_PATH) imagemagick/recoilmagick.c recoil.c -shared $(MAGICK_LDFLAGS) -ldl $(MAGICK_LIBS) -o $@
 else
 	@echo "\nDetected ImageMagick version $(MAGICK_VERSION) on your system."
 	@echo "To build RECOIL coder for ImageMagick,"
@@ -43,7 +43,7 @@ formats.h: formats.h.xsl formats.xml
 	$(CITO) -o $*.c $<
 
 benchmark: benchmark.c recoil-stdio.c recoil-stdio.h recoil.c recoil.h
-	$(CC) $(CFLAGS) benchmark.c recoil-stdio.c recoil.c -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) benchmark.c recoil-stdio.c recoil.c -o $@
 
 clean:
 	rm -f recoil2png imagemagick/recoil.so imagemagick/coder.xml.new formats.h recoil-mime.xml benchmark Xrecoil.usr
-- 
2.35.1

