diff -Nuar crtmpserver-717.old//builders/cmake/CMakeLists.txt crtmpserver-717/builders/cmake/CMakeLists.txt
--- crtmpserver-717.old//builders/cmake/CMakeLists.txt	2012-01-13 18:41:07.000000000 +0600
+++ crtmpserver-717/builders/cmake/CMakeLists.txt	2012-03-31 01:01:11.000000000 +0600
@@ -123,7 +123,7 @@
 #ADD_DEFINITIONS(-DUSE_MEM_POOL)
 #ADD_DEFINITIONS(-DDEBUG_MEM_POOL)
 
-SET(GENERIC_FLAGS "-Wall -Werror")
+SET(GENERIC_FLAGS "-Wall")
 
 #add 64 bit support for 32 but platforms
 ADD_DEFINITIONS(-D__USE_FILE_OFFSET64)
