diff --git a/Source/FreeImage/PluginPNG.cpp b/Source/FreeImage/PluginPNG.cpp
index 71db40d..4061cff 100644
--- a/Source/FreeImage/PluginPNG.cpp
+++ b/Source/FreeImage/PluginPNG.cpp
@@ -37,7 +37,8 @@
 
 // ----------------------------------------------------------
 
-#include "../LibPNG/png.h"
+#define PNG_iTXt_SUPPORTED
+#include <png.h>
 
 // ----------------------------------------------------------
 
